libsolidity/formal/SymbolicTypes.cpp

```diff
@@ -458,6 +494,29 @@ smtutil::Expression zeroValue(frontend::TypePointer const& _type)
 	return 0;
 }
 
+bool isSigned(TypePointer const& _type)
+{
+	solAssert(smt::isNumber(*_type), "");
+	bool isSigned = false;
+	if (auto const* numberType = dynamic_cast<RationalNumberType const*>(_type))
+		isSigned |= numberType->isNegative();
```

Comment:
What was the point of doing this and not just `return numberType->isNegative();`?

Reply:
Not sure, will check and potentially open another PR.